quinn/append_if_schema_identical.py

```diff
@@ -0,0 +1,33 @@
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType, StructField, IntegerType, StringType, DoubleType
+
+
+class SchemaMismatchError(ValueError):
+    """raise this when there's a schema mismatch between source & target schema"""
+
+
+def append_if_schema_identical(source_df: DataFrame, target_df: DataFrame):
+    # Retrieve the schemas of the source and target dataframes
+    source_schema = source_df.schema
+    target_schema = target_df.schema
+
+    # Convert the schemas to a list of tuples
+    source_schema_list = [(field.name, str(field.dataType)) for field in source_schema]
+    target_schema_list = [(field.name, str(field.dataType)) for field in target_schema]
+
+    unmatched_cols = [col for col in source_schema_list if col not in target_schema_list]
+
+    print(source_schema_list)
+    print(type(source_schema_list[0][0]))
+    print(type(target_schema_list[0][0]))
+    print(target_schema_list)
+    error_message = f"The schemas of the source and target dataframes are not identical.From source schema column {unmatched_cols} is missing in target schema"
+    expected_schema = target_schema_list
+    # Compare the two schema lists
+    if source_schema_list == target_schema_list:
```

Comment:
In my understanding if two tables have the same set of fields but in different order we are still able to make append: all that we need to do is just re-order columns of table that we want to append.

Reply:
Imagine the situation when we have table A and table B.
Table A has schema:
```
StructType([StructField("col1", IntegerType()), StructField("col2", StringType())])
```
Table B has schema:
```
StructType([StructField("col2", StringType()), StructField("col1", IntegerType())])
```

In my understanding we can append because schemas are identical and the only difference here is the order of columns/
@MrPowers what do you think? How should we manage such a case?